core/src/main/java/inetsoft/uql/viewsheet/Viewsheet.java

```diff
@@ -2391,22 +2391,15 @@ public synchronized Assembly[] layout(String[] moved) {
       List<Assembly> list = new ArrayList<>();
 
       for(Assembly assembly : getAssemblies()) {
-         if(assembly.getAssemblyType() != VIEWSHEET_ASSET &&
-            !(assembly instanceof ContainerVSAssembly))
+         if(!(assembly instanceof ContainerVSAssembly))
          {
             continue;
          }
 
          Assembly[] arr;
 
-         if(assembly instanceof Viewsheet) {
-            Viewsheet tvs = (Viewsheet) assembly;
-            arr = tvs.layout();
-         }
-         else {
-            ContainerVSAssembly cassembly = (ContainerVSAssembly) assembly;
-            arr = cassembly.layout();
-         }
+         ContainerVSAssembly cassembly = (ContainerVSAssembly) assembly;
```

Comment:
This can be replaced by making the if statement:

`if(!(assembly instanceof ContainerVSAssembly cassembly)) {`